Gaussian Probe optimization (#28872)
idaholab/moose: framework/doc/content/source/materials/ReporterOffsetFunctionMaterial.md

# ReporterOffsetFunctionMaterial

!syntax description /Materials/ReporterOffsetFunctionMaterial

> This creates a material that is the sum of a function shifted by a set of points contained in a reporter.  The reporter can contain points by specifying [!param](/Materials/ReporterOffsetFunctionMaterial/point_name) or by three seperate reporters containing the (x,y,z) coordinates of the points using [!param](/Materials/ReporterOffsetFunctionMaterial/x_coord_name), [!param](/Materials/ReporterOffsetFunctionMaterial/y_coord_name), and [!param](/Materials/ReporterOffsetFunctionMaterial/z_coord_name). This can be useful for creating a field containing multiple sources, as shown in [offset].

Reviewer: ```suggestion
This creates a material that is the sum of functions shifted by a set of points contained in a reporter. The reporter can contain points by specifying [!param](/Materials/ReporterOffsetFunctionMaterial/point_name) or by three separate reporters containing the (x,y,z) coordinates of the points using [!param](/Materials/ReporterOffsetFunctionMaterial/x_coord_name), [!param](/Materials/ReporterOffsetFunctionMaterial/y_coord_name), and [!param](/Materials/ReporterOffsetFunctionMaterial/z_coord_name). This can be useful for creating a field containing multiple sources, as shown in [offset].
```

Reviewer: This first sentence is redundant with the class description. I would delete and focus this paragraph on how to specify the points 

Add the equation for the sum so people can see the minus

> # ReporterOffsetFunctionMaterial

Reviewer: ```suggestion
# ReporterOffsetFunctionSumMaterial
```
I had missed they perform a sum of function

I would move these to the optimization module as well. I dont see who else would use this it's very specific

Reviewer: yeah let's move it all to the optimization module. Should be a couple git mv and remember to re-register from MooseApp to OptimizationApp

Author: I can move it. One use case example is if you have multiple source/nuemann bc terms that are all offset. So like an array of lasers.

!syntax description /Materials/ReporterOffsetFunctionMaterial

This creates a material that is the sum of a function shifted by a set of points contained in a reporter.  The reporter can contain points by specifying [!param](/Materials/ReporterOffsetFunctionMaterial/point_name) or by three seperate reporters containing the (x,y,z) coordinates of the points using [!param](/Materials/ReporterOffsetFunctionMaterial/x_coord_name), [!param](/Materials/ReporterOffsetFunctionMaterial/y_coord_name), and [!param](/Materials/ReporterOffsetFunctionMaterial/z_coord_name). This can be useful for creating a field containing multiple sources, as shown in [offset] and described in the equation below.

> ## $\tilde f(\mathbf{x}) = \sum_{i=1}^{n} \left(f(\mathbf{x} - \mathbf{p}_{offset}) \right)$

Reviewer: ```suggestion
!equation
\tilde f(\mathbf{x}) = \sum_{i=1}^{n} \left(f(\mathbf{x} - \mathbf{p}_{offset}) \right)
```
using a subsection looks weird in the index
